open_lm/datapreprocess/ray/tokenize_shuffle.py

```diff
@@ -484,7 +510,24 @@ def main(args):
             "counter": counter,
         },
         batch_format="pandas",
-    ).count()
+    )
```

Comment:
Instead of a count, do a .take_all()  which should return O(shards) files which is the size of the manifest and then just write directly to s3 using boto3. It will be cleaner, you can even write the manifest to disk
```
import boto3
import json

def upload_jsonl_to_s3(bucket_name, s3_key, data):

    s3_client = boto3.client('s3')


    jsonl_content = '\n'.join(json.dumps(record) for record in data)


    response = s3_client.put_object(
        Bucket=bucket_name,
        Key=s3_key,
        Body=jsonl_content
    )


    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
        print("Upload successful.")
    else:
        print("Upload failed.")
data = [{'key1': 'value1'}, {'key2': 'value2'}]
upload_jsonl_to_s3('my-bucket', 'path/to/myfile.jsonl', data)


```

Reply:
Thanks for the suggestion! Changed this.